src/test/java/at/ac/tuwien/kr/alpha/common/depgraph/ComponentGraphTest.java

```diff
@@ -0,0 +1,265 @@
+import org.junit.Assert;
+import org.junit.Test;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import at.ac.tuwien.kr.alpha.Alpha;
+import at.ac.tuwien.kr.alpha.common.Predicate;
+import at.ac.tuwien.kr.alpha.common.Program;
+import at.ac.tuwien.kr.alpha.common.depgraph.ComponentGraph.SCComponent;
+import at.ac.tuwien.kr.alpha.grounder.structure.ProgramAnalysis;
+
+public class ComponentGraphTest {
+
+	private static final Logger LOGGER = LoggerFactory.getLogger(ComponentGraphTest.class);
+
+	private static void assertNodesMatchStratumNodes(List<SCComponent> stratum, Node... nodes) {
+		ComponentGraphTest.assertNodesMatchStratumNodes(stratum, Arrays.asList(nodes));
+	}
+
+	private static void assertNodesMatchStratumNodes(List<SCComponent> stratum, Iterable<Node> nodes) {
+		List<Node> allNodesInStratum = new ArrayList<>();
+		for (SCComponent comp : stratum) {
+			for (Node compNode : comp.getNodes()) {
+				allNodesInStratum.add(compNode);
+			}
+		}
+
+		for (Node n : nodes) {
+			Assert.assertTrue(n.toString() + " is not contained in stratum, but should be!", allNodesInStratum.contains(n));
+		}
+		boolean isContained;
+		for (Node n : allNodesInStratum) {
+			isContained = false;
+			for (Node n2 : nodes) {
+				if (n2.equals(n)) {
+					isContained = true;
+				}
+			}
+			Assert.assertTrue(n.toString() + " contained in stratum and not in node list, but should be!", isContained);
+		}
+	}
+
+	private static List<Node> extractNodesFromStratum(List<SCComponent> stratum) {
+		List<Node> retVal = new ArrayList<>();
+		for (SCComponent comp : stratum) {
+			retVal.addAll(comp.getNodes());
+		}
+		return retVal;
+	}
+
+	@Test
+	public void stratifyOneRuleTest() throws IOException {
+		Alpha system = new Alpha();
+		Program prog = system.readProgramString("a :- b.", null);
+		ProgramAnalysis pa = new ProgramAnalysis(prog);
+		DependencyGraph dg = pa.getDependencyGraph();
+		ComponentGraph cg = ComponentGraph.fromDependencyGraph(dg);
+		Map<Integer, List<SCComponent>> strata = cg.calculateStratification();
+
+		Assert.assertEquals(1, strata.size());
+	}
+
+	@Test
+	public void stratifyTwoRulesTest() throws IOException {
+		Alpha system = new Alpha();
+		StringBuilder bld = new StringBuilder();
+		bld.append("b :- a.").append("\n");
+		bld.append("c :- b.").append("\n");
+		Program prog = system.readProgramString(bld.toString(), null);
+		ProgramAnalysis pa = new ProgramAnalysis(prog);
+		DependencyGraph dg = pa.getDependencyGraph();
+		ComponentGraph cg = ComponentGraph.fromDependencyGraph(dg);
+		Map<Integer, List<SCComponent>> strata = cg.calculateStratification();
+
+		Assert.assertEquals(1, strata.size());
+		List<SCComponent> stratum0 = strata.get(0);
+		Assert.assertEquals(3, stratum0.size());
+	}
+
+	@Test
+	public void stratifyWithNegativeDependencyTest() throws IOException {
+		Alpha system = new Alpha();
+		StringBuilder bld = new StringBuilder();
+		bld.append("b :- a.").append("\n");
```

Comment:
You can actually use `+` for strings here, since the compiler will turn it into `StringBuilder` usage automatically (or even statically evaluate it and build one large `String` object) .